src/plugins/vis_type_vislib/public/vislib/components/legend/legend_item.tsx

```diff
@@ -42,9 +42,14 @@ import {
   EuiPopoverProps,
   EuiButtonGroup,
   EuiButtonGroupOptionProps,
+  euiPaletteColorBlind,
 } from '@elastic/eui';
 
-import { legendColors, LegendItem } from './models';
+import { LegendItem } from './models';
+
+// starting from the default categorical colors, we generate 6 additional variants,
+// 3 lighter and 3 darker
+const legendColors = euiPaletteColorBlind({ rotations: 7, direction: 'both' });
```

Comment:
Why do we need to set `direction` here?

Reply:
By default, `direction` is `'lighter'`, which means that we wouldn't get the darker colors.

Reply:
Ah I see! So how does it gets divided to 3 lighter and 3 darker?

Reply:
`direction: 'both'` basically generates a palette for each of the 10 base colors that goes from white to the color to black and chops off white and black. Setting `direction` to either `'lighter'` or `'darker'` basically chops off the lighter or darker end of the palette, so setting it to `'both'` just keeps the whole palette.

Reply:
Makes sense!